plutus-errors/src/TH/GenDocs.hs

```diff
@@ -0,0 +1,30 @@
+{-# LANGUAGE TemplateHaskell #-}
+{-# OPTIONS_HADDOCK hide #-}
+module TH.GenDocs (genDocs) where
+
+import           Data.List
+import qualified Data.Text.Prettyprint.Doc as PP
+import           ErrorCode
+import           Errors
+import           Language.Haskell.TH       as TH
+import           TH.GenCodes
+
+-- | Generate haddock documentation for all errors and their codes,
+-- by creating type-synonyms to lifted dataconstructors using a DataKinds trick.
+genDocs :: Q [TH.Dec]
+genDocs = let allCodes = $(genCodes allErrors)
+          in case findDuplicates allCodes of
+                 []   -> pure $ fmap mkTySyn (zip allErrors allCodes)
+                 -- Fail at compile time if duplicate error codes are found.
+                 dups -> fail $ "ErrorCode instances have some duplicate error-code numbers: " ++ (show $ PP.pretty dups)
+
+-- | An alias (type-synonym) for a given error,
+-- using naming scheme "E+error-code".
+mkTySyn :: (TH.Name,E) -> TH.Dec
+mkTySyn (err, code) =
+    let aliasName = mkName $ show $ PP.pretty code
+    in TySynD aliasName [] $ ConT err
+
+-- | find the duplicate occurences in a list
```

Comment:
Seems okay. I'd probably do something like convert to a set and check whether the size of the set was the same as the size of the list.